### DISCUSSION TITLE
how can we have q-drawer in its default form (true) and on some other pages (false) or in mini mode??

### DISCUSSION THREAD

**Discussion by FazilNusrat** [quasarframework/quasar#13343] [General - Components / Directives / etc] [answered]
Hi dear seniors, I want to have q-drawer in some specific pages, because I need to display full page in some of them.
kindly tell me some better way
I tried more and I did'nt succeed yet.

I also tried this block of code in created()
this.$q.leftDrawerOpen = false;
but not worked.

Note: I do not mean that I want to hide it like the login page, I mean it should opens again by clicking the menu button.

**Comment by MilosPaunovic** [comment, marked as answer]
You need to use `v-model` to control the state of your `QDrawer`, like in the example from the docs:
[https://codepen.io/milospaunovic/pen/LYQVqvE?editors=101](https://codepen.io/milospaunovic/pen/LYQVqvE?editors=101)

So you could use `created` lifecycle hook to open it when needed and close it in `beforeDestroy`.
Try to create something and if you get stuck it's the best if you provide some kind of code repro, for someone to be able to help you.

**Reply by FazilNusrat** [reply]
> You need to use `v-model` to control the state of your `QDrawer`, like in the example from the docs: https://codepen.io/milospaunovic/pen/LYQVqvE?editors=101
> 
> So you could use `created` lifecycle hook to open it when needed and close it in `beforeDestroy`. Try to create something and if you get stuck it's the best if you provide some kind of code repro, for someone to be able to help you.


First of all, thank you for replying Mr. @MilosPaunovic 
I completely did the all required functionalities like v-model in Layout.vue and by default it's true in all pages 
but I want to change the status of th QDrawer to false in specific page, unfortunately it does not defined globally to control it in another specific pages

**Reply by MilosPaunovic** [reply]
Are you using any state management library, like Vuex or Pinia? If so, you could move the value which you're using to toggle `QDrawer` there, and create a simple mutation which you can call from the any component you need it to.

**Reply by FazilNusrat** [reply]
Unfortunately I did not use any state management and is it possible to use some global functions in script like (this.$q.leftDrawerOpen = false;) or I have to install some state management?

**Reply by MilosPaunovic** [reply]
You could pass and listen for events, it might be a better solution for your particular need.
[https://stackoverflow.com/questions/63471824/vue-js-3-event-bus](https://stackoverflow.com/questions/63471824/vue-js-3-event-bus)

But adding state management couldn't hurt, and it would be much easier solution.

**Reply by FazilNusrat** [reply]
Yeah, in QDrawer API they mentioned the show and hide events, I have to learn and apply it, I think this will be the better way

**Reply by FazilNusrat** [reply]
So it's good to work on state management, because it's new for me and it will help me for further tasks
Thank you in advance Mr. @MilosPaunovic
